js/src/components/paid-ads/convertToAssetGroupUpdateBody.js

```diff
@@ -0,0 +1,91 @@
+/**
+ * Internal dependencies
+ */
+import { ASSET_KEY, ASSET_FORM_KEY } from '.~/constants';
+
+/**
+ * @typedef {import('.~/components/types.js').AssetGroupFormValues} AssetGroupFormValues
+ * @typedef {import('.~/data/types.js').AssetEntityGroup} AssetEntityGroup
+ * @typedef {import('.~/data/types.js').AssetOperations} AssetOperations
+ * @typedef {import('.~/data/types.js').AssetEntityGroupUpdateBody} AssetEntityGroupUpdateBody
+ */
+
+/**
+ * Differences the assets form values against the existing asset entity group and
+ * returns the updating operations of the asset group.
+ *
+ * @param  {AssetEntityGroup} assetGroup The existing asset entity group as the base data for comparing the differences.
+ * @param  {AssetGroupFormValues} values The assets form values to be compared.
+ * @return {AssetOperations[]} The creation and deletion operations for updating the asset group.
+ */
+export function diffAssetOperations( assetGroup, values ) {
+	const operations = [];
+	const assets = assetGroup.assets;
+
+	function pushDeletions( key, ...assetEntities ) {
+		const deletions = assetEntities.map( ( assetEntity ) => ( {
+			...assetEntity,
+			// Setting the `content` to null indicates the asset deletion operation.
+			content: null,
+			field_type: key,
+		} ) );
+		operations.push( ...deletions );
+	}
+
+	Object.values( ASSET_KEY ).forEach( ( key ) => {
```

Comment:
Same as my comment in previous PRs. It was confusing that we want to iterate over the ASSET_KEYs but name is still singular.

Reply:
Unfortunately, I'm not pretty sure why seeing `Object.values( ASSET_KEY )` is confusing. The only purpose of the `Object.values` function is to accept an object type parameter and return an array of a given object's own enumerable property values.

> It was confusing that we want to iterate over the ASSET_KEYs but name is still singular.

In other words by the same token, `Object.values( ASSET_KEYS )` could be still considered confusing because "why it needs to iterate over an array?"

And It's just the reason why `ASSET_KEYS` might be inferred as the array type and may try to `ASSET_KEYS.forEach()` first then get the "{}.forEach is not a function" error.

Reply:
I don't know how to explain it in other words. For me seems obvious that we are iterating over a set of (plural) keys and not just a single Key. Not a blocker tho since it's a naming issue. But I still disagree ASSET_KEY is correct